engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java

```diff
@@ -405,29 +405,45 @@ private void validateSortField(SortField sortField, DocumentFieldMetadata fieldM
 			}
 	}
 
-	private void validateNumericSortField(SortField sortField, NumericEncodingType numericEncodingType) {
+	private void validateNumericSortField(SortField sortField, NumericEncodingType indexNumericEncodingType) {
+		final NumericEncodingType sortNumericEncodingType;
 		switch ( sortField.getType() ) {
 			case BYTES:
 			case INT:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.INTEGER );
+				sortNumericEncodingType = NumericEncodingType.INTEGER;
 				break;
 			case LONG:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.LONG );
+				sortNumericEncodingType = NumericEncodingType.LONG;
 				break;
 			case DOUBLE:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.DOUBLE );
+				sortNumericEncodingType = NumericEncodingType.DOUBLE;
 				break;
 			case FLOAT:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.FLOAT );
+				sortNumericEncodingType = NumericEncodingType.FLOAT;
 				break;
 			default:
-				throw LOG.sortTypeDoesNotMatchFieldType( String.valueOf( sortField.getType() ), String.valueOf( numericEncodingType ), sortField.getField() );
+				throw LOG.sortTypeDoesNotMatchFieldType( String.valueOf( sortField.getType() ),
+						String.valueOf( indexNumericEncodingType ), sortField.getField() );
 		}
+		if ( NumericEncodingType.UNKNOWN.equals( indexNumericEncodingType ) ) {
+			/*
+			 * The actual encoding type is unknown, so we can't validate more.
+			 * This happens most notably when using custom numeric field bridges.
+			 * We can't ask the user to add more metadata, since there is no annotation for
```

Comment:
Interestingly, it does not. See https://github.com/yrodiere/hibernate-search/commit/c386e678cf36e0617f744a7bd272925ce11437b4 and https://travis-ci.org/yrodiere/hibernate-search/builds/153335179
I see what you mean, though: maybe I was simply doing it wrong, and it happens that there also is an entirely different bug when doing it right.
Regarding this pull request, I think that even if remove my fix (because it's inappropriate), maybe I should still offer a more explicit error message and consider it's a fix for HSEARCH-2325.
I'll work on this tomorrow (I'll try to see where the bug lies exactly)